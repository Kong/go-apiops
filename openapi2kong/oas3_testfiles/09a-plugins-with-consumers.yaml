# x-kong-plugin-<name> can carry a consumer id. Those plugins must be added
# on the top-level of the output because they have 2 foreign keys.
#
# the top-level plugins array should be lexically sorted by plugin-name + ID

openapi: '3.0.0'
info:
  title: Simple API overview
  version: v2
servers:
  - url: https://server1.com/

x-kong-plugin-request-termination:
  # name will be determined from the tag
  # name: request-termination;
  config:
    status_code: 403
    message: So long and thanks for all the fish!
  # foreign keys to consumer
  consumer: johndoe1

paths:
  /path1:
    get:
      # This operation should not get a plugin, but rely on the document level
      # service-entity based plugin
      operationId: uses-doc-plugin
      summary: List API versions
      responses:
        '200':
          description: |-
            200 response
  /path2:
    # This plugin gets attached to a route, overriding the document service-entity
    # attached plugin
    x-kong-plugin-request-termination:
      config:
        status_code: 403
        message: The answer to life, the universe, and everything!
      # foreign keys to consumer
      consumer: johndoe2
    get:
      # gets the plugin from the path-level, attached on route-entity
      operationId: uses-path-plugin
      summary: List API versions
      responses:
        '200':
          description: |-
            200 response
    post:
      # This plugin gets attached to a route, taking precedence over the path-level
      # plugin. On the route it will override the document level service-entity
      # attached plugin
      x-kong-plugin-request-termination:
        config:
          status_code: 403
          message: For a moment, nothing happened. Then, after a second or so, nothing continued to happen.
        # foreign keys to consumer
        consumer: johndoe3
      operationId: uses-ops-plugin
      summary: List API versions
      responses:
        '200':
          description: |-
            200 response
