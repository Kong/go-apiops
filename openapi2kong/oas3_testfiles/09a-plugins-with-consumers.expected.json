{
  "_format_version": "3.0",
  "plugins": [
    {
      "config": {
        "message": "The answer to life, the universe, and everything!",
        "status_code": 403
      },
      "consumer": "johndoe2",
      "id": "aa56031e-7155-599f-a9e9-93e6b271ba58",
      "name": "request-termination",
      "route": "simple-api-overview_uses-path-plugin",
      "tags": [
        "OAS3_import",
        "OAS3file_09a-plugins-with-consumers.yaml"
      ]
    },
    {
      "config": {
        "message": "For a moment, nothing happened. Then, after a second or so, nothing continued to happen.",
        "status_code": 403
      },
      "consumer": "johndoe3",
      "id": "ead16074-ccb0-52dd-9f56-4193529e8ffa",
      "name": "request-termination",
      "route": "simple-api-overview_uses-ops-plugin",
      "tags": [
        "OAS3_import",
        "OAS3file_09a-plugins-with-consumers.yaml"
      ]
    },
    {
      "config": {
        "message": "So long and thanks for all the fish!",
        "status_code": 403
      },
      "consumer": "johndoe1",
      "id": "ef93d30f-dc7d-581b-8b1b-3f8942a0d171",
      "name": "request-termination",
      "service": "simple-api-overview",
      "tags": [
        "OAS3_import",
        "OAS3file_09a-plugins-with-consumers.yaml"
      ]
    }
  ],
  "services": [
    {
      "host": "server1.com",
      "id": "0907c4ab-d9e4-5d21-813b-c57a97eeaad9",
      "name": "simple-api-overview",
      "path": "/",
      "plugins": [],
      "port": 443,
      "protocol": "https",
      "routes": [
        {
          "id": "d020c736-e762-5dad-8415-1684ee4f2061",
          "methods": [
            "GET"
          ],
          "name": "simple-api-overview_uses-doc-plugin",
          "paths": [
            "~/path1$"
          ],
          "plugins": [],
          "regex_priority": 200,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_09a-plugins-with-consumers.yaml"
          ]
        },
        {
          "id": "ef2ca083-29b3-5d7b-87c5-e4315d830c33",
          "methods": [
            "GET"
          ],
          "name": "simple-api-overview_uses-path-plugin",
          "paths": [
            "~/path2$"
          ],
          "plugins": [],
          "regex_priority": 200,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_09a-plugins-with-consumers.yaml"
          ]
        },
        {
          "id": "f9c8a7d7-3518-5c25-b66b-40943e59f91b",
          "methods": [
            "POST"
          ],
          "name": "simple-api-overview_uses-ops-plugin",
          "paths": [
            "~/path2$"
          ],
          "plugins": [],
          "regex_priority": 200,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_09a-plugins-with-consumers.yaml"
          ]
        }
      ],
      "tags": [
        "OAS3_import",
        "OAS3file_09a-plugins-with-consumers.yaml"
      ]
    }
  ],
  "upstreams": []
}
